Origin: https://github.com/ruby/ruby/commit/d8d19683b62cc8e9254e651acd89a6cdfa3a0f54
Reviewed-by: Sylvain Beucler <beuc@debian.org>
Last-Update: 2024-07-16

This fixes the following flaky failure:
  1) Failure:
  TestRipper::Generic#test_parse_files [/usr/src/ruby/ruby/test/ripper/test_files.rb:8]:
  assert_separately failed with error message
  pid 20444 exit 0
  | /usr/src/ruby/ruby/test/rubygems/test_gem_server.rb:391: warning: character class has duplicated range
  | /usr/src/ruby/ruby/test/rubygems/test_gem_server.rb:446: warning: character class has duplicated range
  | /usr/src/ruby/ruby/test/rubygems/test_gem_server.rb:473: warning: character class has duplicated range
  | /usr/src/ruby/ruby/test/rubygems/test_gem_server.rb:500: warning: character class has duplicated range

From d8d19683b62cc8e9254e651acd89a6cdfa3a0f54 Mon Sep 17 00:00:00 2001
From: nagachika <nagachika@b2dd03c8-39d4-4d8f-98ff-823fe69b080e>
Date: Fri, 16 Feb 2018 16:04:21 +0000
Subject: fix regexp literal warning.

test/rubygems/test_gem_server.rb: eliminate duplicated character class warning.
[Bug #14481]

git-svn-id: svn+ssh://ci.ruby-lang.org/ruby/trunk@62436 b2dd03c8-39d4-4d8f-98ff-823fe69b080e
---
 test/rubygems/test_gem_server.rb | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: ruby/test/rubygems/test_gem_server.rb
===================================================================
--- ruby.orig/test/rubygems/test_gem_server.rb
+++ ruby/test/rubygems/test_gem_server.rb
@@ -388,7 +388,7 @@ class TestGemServer < Gem::TestCase
     #
     #
     #  <a href="." title=".">[www]</a>
-    regex_match = /xsshomepagegem 1<\/b>[\n\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/xsshomepagegem-1\/">\[rdoc\]<\/a>)[\n\s]+<a href="\." title="\.">\[www\]<\/a>/
+    regex_match = /xsshomepagegem 1<\/b>[\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/xsshomepagegem-1\/">\[rdoc\]<\/a>)[\s]+<a href="\." title="\.">\[www\]<\/a>/
     assert_match regex_match, @res.body
   end
 
@@ -443,7 +443,7 @@ class TestGemServer < Gem::TestCase
     #
     #
     #  <a href="." title=".">[www]</a>
-    regex_match = /invalidhomepagegem 1<\/b>[\n\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/invalidhomepagegem-1\/">\[rdoc\]<\/a>)[\n\s]+<a href="\." title="\.">\[www\]<\/a>/
+    regex_match = /invalidhomepagegem 1<\/b>[\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/invalidhomepagegem-1\/">\[rdoc\]<\/a>)[\s]+<a href="\." title="\.">\[www\]<\/a>/
     assert_match regex_match, @res.body
   end
 
@@ -470,7 +470,7 @@ class TestGemServer < Gem::TestCase
     assert_equal 200, @res.status
     assert_match 'validhomepagegemhttp 1', @res.body
 
-    regex_match = /validhomepagegemhttp 1<\/b>[\n\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/validhomepagegemhttp-1\/">\[rdoc\]<\/a>)[\n\s]+<a href="http:\/\/rubygems\.org" title="http:\/\/rubygems\.org">\[www\]<\/a>/
+    regex_match = /validhomepagegemhttp 1<\/b>[\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/validhomepagegemhttp-1\/">\[rdoc\]<\/a>)[\s]+<a href="http:\/\/rubygems\.org" title="http:\/\/rubygems\.org">\[www\]<\/a>/
     assert_match regex_match, @res.body
   end
 
@@ -497,7 +497,7 @@ class TestGemServer < Gem::TestCase
     assert_equal 200, @res.status
     assert_match 'validhomepagegemhttps 1', @res.body
 
-    regex_match = /validhomepagegemhttps 1<\/b>[\n\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/validhomepagegemhttps-1\/">\[rdoc\]<\/a>)[\n\s]+<a href="https:\/\/rubygems\.org" title="https:\/\/rubygems\.org">\[www\]<\/a>/
+    regex_match = /validhomepagegemhttps 1<\/b>[\s]+(<span title="rdoc not installed">\[rdoc\]<\/span>|<a href="\/doc_root\/validhomepagegemhttps-1\/">\[rdoc\]<\/a>)[\s]+<a href="https:\/\/rubygems\.org" title="https:\/\/rubygems\.org">\[www\]<\/a>/
     assert_match regex_match, @res.body
   end
 
